### PR TITLE
add code for moving user creation date if transaction before exist

### PR DESCRIPTION
**Pull request by einhornimmond**
### Issues

- None

### Todo
[x] Test with Gradido Node

### CODE REVIEW

**Review by Mogge — COMMENTED** [review]
Looks good!

**Review by Mogge — CHANGES_REQUESTED** [review]
I suggest to rename the migration file to `0041-...`
So we can merge this one. To get it pass the CI tests, you have to update the `DB_VERSION` in `backend/src/config/index.ts`

**Comment by einhornimmond** [comment]
> I suggest to rename the migration file to `0041-...` So we can merge this one. To get it pass the CI tests, you have to update the `DB_VERSION` in `backend/src/config/index.ts`

This get's the 42 because the 41 is already used from #2029, so than we can merge this one after merging the #2029

**Review by Mogge — APPROVED** [review]
I really suggest to merge this first.

File: backend/src/config/index.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ Decimal.set({
 })
 
 const constants = {
-  DB_VERSION: '0040-add_contribution_link_id_to_user',
+  DB_VERSION: '0042-move_users_creation_date',
```
Comment (Mogge) [inline backend/src/config/index.ts:13]:
```suggestion
  DB_VERSION: '0041-move_users_creation_date',
```

File: database/migrations/0042-move_users_creation_date.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
```
Comment (Mogge) [inline database/migrations/0042-move_users_creation_date.ts:1]: Change file name to `0041`

**Review by Mogge — CHANGES_REQUESTED** [review]

File: database/migrations/0042-move_users_creation_date.ts (hunk 2 above)
Comment (einhornimmond) [reply database/migrations/0042-move_users_creation_date.ts]: > I really suggest to merge this first.

ok, should also work in this order.

**Review by Mogge — APPROVED** [review]

**Pull request merged**